### PR TITLE
Add option to zip content to redis and to force redis updates

### PR DESCRIPTION
**Pull request by adi-rose**

### CODE REVIEW

File: cloud_utils/cache/stores/redis_sync.py [hunk 1]
```diff
@@ -35,21 +36,24 @@ def get_item(key: str):
...
         try:
+            if type(result) is bytes:
+                return result
             return json.loads(result)
         except ValueError:  # Key contents are malformed (will force key to update).
             logging.error(f"Malformed key detected: {key} in {name}.")
             raise KeyError
 
     def set_item(key: str, value):
+        value = json.loads(value) if type(value) is not bytes else value
```
Comment (JonSc8) [inline cloud_utils/cache/stores/redis_sync.py:47]: this should be json.dumps no?

as this is the return value in line 47

unless i'm missing something

Comment (adi-rose) [reply cloud_utils/cache/stores/redis_sync.py:47]: yes, should totally be json.dumps! got the two confused

File: cloud_utils/cache/stores/redis_sync.py [hunk 2]
```diff
@@ -35,21 +36,24 @@ def get_item(key: str):
             logging.debug(f"{key} is not in {name}")
             raise KeyError
         try:
```
Comment (nitzo) [inline cloud_utils/cache/stores/redis_sync.py:38]: I think this should be an explicit parameter and not use a heuristic to determine if you want to dumps and loads or not. 
Another option, maybe just use the decode/enode functions now that we have them ? wdyt ?

**Review by nitzo — APPROVED** [review]

File: cloud_utils/cache/stores/redis_sync.py (hunk 2 above)
Comment (adi-rose) [reply cloud_utils/cache/stores/redis_sync.py:36]: Sounds good! 
I tried it at the beginning when writing with the decode and encode function and had some issues with it so I added an explicit parameter to the function now

**Pull request merged**